libs/libGamedata/gameData/CampaignDescription.cpp

```diff
@@ -34,7 +34,16 @@ CampaignDescription::CampaignDescription(const kaguya::LuaRef& table)
     lua::validatePath(luaFolder);
     mapNames = luaData.getOrDefault("maps", std::vector<std::string>());
 
-    luaData.checkUnused();
+    kaguya::LuaRef entry = table["selectionMap"];
+    if(entry.type() == LUA_TTABLE)
+    {
+        selectionMapData = std::make_optional<SelectionMapInputData>(entry);
+    }
+}
+
+std::optional<SelectionMapInputData> CampaignDescription::getSelectionMapData() const
```

Comment:
No need for copies as sometimes you just want to check IFF there is a value.
```suggestion
const std::optional<SelectionMapInputData>& CampaignDescription::getSelectionMapData() const
```